crates/rover-client/src/query/partial/delete.rs

```diff
@@ -0,0 +1,160 @@
+use crate::blocking::StudioClient;
+use crate::RoverClientError;
+use graphql_client::*;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/query/partial/delete.graphql",
+    schema_path = "schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. delete_service_mutation
+pub struct DeleteServiceMutation;
+type RawMutationResponse = delete_service_mutation::DeleteServiceMutationServiceRemoveImplementingServiceAndTriggerComposition;
+
+/// this struct contains all the info needed to print the result of the delete.
+/// `updated_gateway` is true when composition succeeds and the gateway config
+/// is updated for the gateway to consume. `composition_errors` is just a list
+/// of strings for when there are composition errors as a result of the delete.
+#[derive(Debug, PartialEq)]
+pub struct DeleteServiceResponse {
+    pub updated_gateway: bool,
+    pub composition_errors: Option<Vec<String>>,
+}
+
+/// The main function to be used from this module. This function fetches a
+/// schema from apollo studio and returns it in either sdl (default) or json format
+pub fn run(
+    variables: delete_service_mutation::Variables,
+    client: &StudioClient,
+) -> Result<DeleteServiceResponse, RoverClientError> {
+    let response_data = client.post::<DeleteServiceMutation>(variables)?;
+    let data = get_delete_data_from_response(response_data)?;
+    Ok(build_response(data))
+}
+
+fn get_delete_data_from_response(
+    response_data: delete_service_mutation::ResponseData,
+) -> Result<RawMutationResponse, RoverClientError> {
+    let service_data = match response_data.service {
+        Some(data) => Ok(data),
+        None => Err(RoverClientError::HandleResponse {
+            msg: "No service found".to_string(),
+        }),
```

Comment:
this is a case where i would add a new variant to `RoverClientError`. so instead of `Err(RoverClientError::HandleResponse( { msg: "Stringly typed message" }`, I'd do `Err(RoverClientError::NoServiceFound)` with the message you want in the `#[error("message")]` portion

Reply:
That makes sense!